src/main/java/com/springjpa/datajpa/Service/GeoService.java

```diff
@@ -28,49 +33,74 @@
     private String geoDTO;
     @Autowired
     private WebClient webClient;
-    @Autowired
-    StationScrapperService stationScrapperService;
+
     public List<Geo> downloadAllGeos() throws JSONException {
         Iterable<Station> stations = stationScrapperService.getAllStations();
-            stations.forEach(station -> {
-            name=station.getName();
+        stations.forEach(station -> {
+            try {
+                name = station.getName();
                 System.out.println(name);
-            geoDTO = webClient.get()
-                    .uri(uriBuilder -> uriBuilder
-                            .queryParam("query","Warszawa")
-                            .build())
-                    .retrieve()
-                    .bodyToMono(String.class)
-                    .block();
-                try {
-                    jsonObject = new JSONObject(geoDTO);
-                } catch (JSONException e) {
-                    e.printStackTrace();
-                }
-                try {
-                    jsonArray = jsonObject.getJSONArray("data");
-                } catch (JSONException e) {
-                    e.printStackTrace();
-                }
-                try {
-                    jsonObject = jsonArray.getJSONObject(0);
-                } catch (JSONException e) {
-                    e.printStackTrace();
-                }
-                try {
-                    latitude = jsonObject.getDouble("latitude");
-                } catch (JSONException e) {
-                    e.printStackTrace();
+                ApiResponseDTO response = webClient.get()
+                        .uri(uriBuilder -> uriBuilder
+                                .queryParam("query", "Warszawa")
+                                .build())
+                        .retrieve()
+                        .bodyToMono(ApiResponseDTO.class)
+                        .block();
+                if (response != null) {
+                    Geo newGeo = createGeoFromResponse(response);
+                    System.out.println(newGeo);
+                    geoList.add(newGeo);
                 }
-                try {
-                    longitude = jsonObject.getDouble("longitude");
-                } catch (JSONException e) {
-                    e.printStackTrace();
-                }
-                geoList.add(new Geo(longitude,latitude));
-            });
+            } catch (RuntimeException e) {
+                System.out.println("Error while fetching geo data for station " + name + ": " + e.getMessage());
+            }
+
+//            try {
```

Comment:
To Ci zostawiłem, ale jeśli uznasz, że te moje rozwiązanie jest ok, to ja bym wyjebał cały ten zakomentowany kod